lib/Runtime/Library/JavascriptSymbol.cpp

```diff
@@ -242,6 +242,12 @@ namespace Js
 
     BOOL JavascriptSymbol::Equals(JavascriptSymbol* left, Var right, BOOL* value, ScriptContext * requestContext)
     {
+        TypeId typeId = JavascriptOperators::GetTypeId(right);
+        if (typeId != TypeIds_Symbol && typeId != TypeIds_SymbolObject)
+        {
+            right = JavascriptConversion::ToPrimitive(right, JavascriptHint::None, requestContext);
```

Comment:
Can we update typeId with the new type of right inside this if? Then we can switch on typeId which would save us a call to GetTypeId on the Symbol or SymbolObject case.
```c++
if (typeId != TypeIds_Symbol && typeId != TypeIds_SymbolObject) 
{  
  right = JavascriptConversion::ToPrimitive(right, JavascriptHint::None, requestContext); 
  typeId = JavascriptOperators::GetTypeId(right); 
}
```